test/gtest/uct/test_peer_failure.cc

```diff
@@ -280,32 +302,35 @@ UCS_TEST_SKIP_COND_P(test_uct_peer_failure, purge_failed_peer,
     send_recv_am(0);
     send_recv_am(1);
 
-    const ucs_time_t loop_end_limit = ucs::get_deadline();
-    const size_t num_pend_sends     = 64ul;
-    const uint64_t send_data        = 0;
+    const size_t num_pend_sends = 64ul;
+    const size_t max_retry_num  = 3ul;
     std::vector<pending_send_request_t> reqs(num_pend_sends);
 
     {
         scoped_log_handler slh(wrap_errors_logger);
 
-        ucs_status_t status;
-        do {
-            status = uct_ep_am_short(ep0(), 0, 0, &send_data,
-                                     sizeof(send_data));
-        } while ((status == UCS_OK) && (ucs_get_time() < loop_end_limit));
+        fill_resources();
 
-        if (status == UCS_OK) {
-            UCS_TEST_SKIP_R("unable to fill the UCT resources");
-        } else if (status != UCS_ERR_NO_RESOURCE) {
-            UCS_TEST_ABORT("AM Short failed with " << ucs_status_string(status));
+        kill_receiver();
+
+        size_t retry_cnt;
+        for (retry_cnt = 0; retry_cnt < max_retry_num; retry_cnt++) {
+            ucs_status_t status = add_pending(ep0(), reqs[0]);
+            if (UCS_OK == status) {
+                break;
+            } else {
+                EXPECT_EQ(UCS_ERR_BUSY, status);
+                fill_resources();
+            }
         }
 
-        kill_receiver();
+        if (retry_cnt == max_retry_num) {
+            UCS_TEST_SKIP_R("unable to add pending requests after "
```

Comment:
looks excessive, because checked in `fill_resources` already in some way

Reply:
@yosefe wdyt?
btw, added one more comment to fix case when AM short fails with error in `fill_resources()` after peer failure was detected

Reply:
@brminich it's different, it means we got UCS_ERR_BUSY too many times from add_pending